client/my-sites/site-settings/form-general.jsx

```diff
@@ -34,69 +37,70 @@ import SiteIconSetting from './site-icon-setting';
 import { FEATURE_NO_BRANDING } from 'lib/plans/constants';
+import {
+	getSiteSettingsSaveRequestStatus,
+	isRequestingSiteSettings,
+	isSavingSiteSettings,
+	getSiteSettings
+} from 'state/site-settings/selectors';
+import { saveSiteSettings } from 'state/site-settings/actions';
+import { removeNotice } from 'state/notices/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import QuerySiteSettings from 'components/data/query-site-settings';
+
+const debug = debugFactory( 'calypso:my-sites:site-settings' );
+
+class SiteSettingsFormGeneral extends Component {
+	state = {};
+
+	getFormSettings( settings ) {
+		if ( ! settings ) {
+			return {};
+		}
 
-const FormGeneral = React.createClass( {
-	displayName: 'SiteSettingsFormGeneral',
-
-	mixins: [ dirtyLinkedState, formBase ],
+		const formSettings = {
+			blogname: settings.blogname,
+			blogdescription: settings.blogdescription,
 
-	getSettingsFromSite( site ) {
-		site = site || this.props.site;
+			lang_id: settings.lang_id,
+			blog_public: settings.blog_public,
+			timezone_string: settings.timezone_string,
+			jetpack_relatedposts_allowed: settings.jetpack_relatedposts_allowed,
+			jetpack_sync_non_public_post_stati: settings.jetpack_sync_non_public_post_stati,
 
-		const settings = {
-			blogname: site.name,
-			blogdescription: site.description,
-			fetchingSettings: site.fetchingSettings
+			amp_is_supported: settings.amp_is_supported,
+			amp_is_enabled: settings.amp_is_enabled
 		};
 
-		if ( site.settings ) {
-			settings.lang_id = site.settings.lang_id;
-			settings.blog_public = site.settings.blog_public;
-			settings.admin_url = site.settings.admin_url;
-			settings.timezone_string = site.settings.timezone_string;
-			settings.jetpack_relatedposts_allowed = site.settings.jetpack_relatedposts_allowed;
-			settings.jetpack_sync_non_public_post_stati = site.settings.jetpack_sync_non_public_post_stati;
-
-			settings.amp_is_supported = site.settings.amp_is_supported;
-			settings.amp_is_enabled = site.settings.amp_is_enabled;
-
-			if ( settings.jetpack_relatedposts_allowed ) {
-				settings.jetpack_relatedposts_enabled = ( site.settings.jetpack_relatedposts_enabled ) ? 1 : 0;
-				settings.jetpack_relatedposts_show_headline = site.settings.jetpack_relatedposts_show_headline;
-				settings.jetpack_relatedposts_show_thumbnails = site.settings.jetpack_relatedposts_show_thumbnails;
-			}
-
-			if ( site.settings.holidaysnow ) {
-				settings.holidaysnow = site.settings.holidaysnow;
-			}
-
-			// handling `gmt_offset` and `timezone_string` values
-			const gmt_offset = site.settings.gmt_offset;
-
-			if (
-				! settings.timezone_string &&
-				typeof gmt_offset === 'string' &&
-				gmt_offset.length
-			) {
-				settings.timezone_string = 'UTC' +
-					( /\-/.test( gmt_offset ) ? '' : '+' ) +
-					gmt_offset;
-			}
+		if ( settings.jetpack_relatedposts_allowed ) {
+			formSettings.jetpack_relatedposts_enabled = ( settings.jetpack_relatedposts_enabled ) ? 1 : 0;
```

Comment:
Code is legacy, though when assigning values of multiple properties, could be more readable with `Object.assign`, i.e.:

```js
if ( settings.jetpack_relatedposts_allowed ) {
	Object.assign( formSettings, {
		jetpack_relatedposts_enabled: ( settings.jetpack_relatedposts_enabled ) ? 1 : 0,
		jetpack_relatedposts_show_headline: settings.jetpack_relatedposts_show_headline,
		jetpack_relatedposts_show_thumbnails: settings.jetpack_relatedposts_show_thumbnails
	} );
}
```